### PR TITLE
docs: upgrade storybook and add docs per chart

### PR DESCRIPTION
**Pull request by rshen91**
## Summary

This PR addresses issue #457 with upgrading storybook to 5.2.8 and enables the functionality to include separate docs per chart story.  

Will add docs to stories in a future PR. 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Proper documentation or storybook story was added for features that require explanation or tutorials~~
~~- [ ] Unit tests were updated or added to match the most common scenarios~~
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
@rshen91 I've check this PR and I've found few issues:
- these 3 dependency are deprecated in storybook 5.2.8, as they are already included into storybook package itself
```
"@types/storybook__addon-info": "^4.1.2",
"@types/storybook__addon-knobs": "5.0.4",
"@types/storybook__react": "^4.0.2",
```
- you should update `@types/storybook__addon-info` to the latest available
- The types of `info addon` are slightly changed: you have to remove the `withInfo` decorator and add the info specification in the `addParameters` arguments inside an `info` object like:
```
addParameters({
  options: {
    theme: create({
      base: 'light',
      brandTitle: 'Elastic Charts',
      brandUrl: 'https://github.com/elastic/elastic-charts',
      brandImage:
        'https://static-www.elastic.co/v3/assets/bltefdd0b53724fa2ce/blt6ae3d6980b5fd629/5bbca1d1af3a954c36f95ed3/logo-elastic.svg',
    }),
    panelPosition: 'right',
    sidebarAnimations: true,
  },
  info: {
    ...the info addon configuration
  },
});
```

- the `changeNumbers` function used for the array issue could be an utility shared across all stories. Also note that you can write it as a `map` instead of a for each

**Comment by rshen91** [comment]
This is awesome feedback, thank you @markov00! I'm still struggling trying to figure out the styles within `addParameters options` vs the `addDecorator` route. Doing the above suggestion makes sense but I  wasn't able to have it apply the styles for infoBody.

 I've moved `changeNumbers()` into the specs.ts file, that makes so much more sense!

**Comment by markov00** [comment]
@rshen91 I've seen the issue with the `Story source`. Looking at the errors in the console seems related to an issue with CSP https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy

For a quick workaround (that just disable partially the CSP) you can change the code in `.storybook/preview-head.html` with the following:
```
<meta
  http-equiv="Content-Security-Policy"
  content="default-src 'self'; style-src 'unsafe-inline';"
/>
```
this will at least allows you to see correctly the source code.

**Comment by rshen91** [comment]
I was seeing some integration tests locally failing but I'm not sure why - in commit 3bb8b4e I changed array to arrayKnobs in another story but if @markov00 or @nickofthyme spot anything that this PR might have introduced to explain the failing visual tests, any pointers would be appreciated!

File: .storybook/preview-head.html [hunk 1]
```diff
@@ -1,4 +1 @@
-<meta
-  http-equiv="Content-Security-Policy"
-  content="default-src 'self'; style-src 'nonce-Pk1rZ1XDlMuYe8ubWV3Lh0BzwrTigJQ='"
-/>
+<meta http-equiv="Content-Security-Policy" content="default-src 'self'; style-src 'unsafe-inline';" />
```
[the remark this answers is not in the document]
Comment (markov00) [reply .storybook/preview-head.html]: fixed to a better CSP here: 4707242

File: package.json [hunk 2]
```diff
@@ -142,7 +140,9 @@
     "webpack-dev-server": "^3.3.1"
   },
   "dependencies": {
+    "@storybook/addon-docs": "^5.2.8",
     "@types/d3-shape": "^1.3.1",
+    "@types/storybook__addon-info": "^5.2.1",
```
[the remark this answers is not in the document]
Comment (markov00) [reply package.json]: fix in 78179d2

File: package.json [hunk 3]
```diff
@@ -121,8 +119,8 @@
     "prettier": "1.16.4",
     "pretty-quick": "^2.0.0",
     "puppeteer": "^1.20.0",
-    "react": "^16.12.0",
-    "react-docgen-typescript-loader": "^3.0.1",
+    "react": "16.10.x",
```
[the remark this answers is not in the document]
Comment (markov00) [reply package.json]: fix in 78179d2

**Pull request merged**